### PR TITLE
(CDAP-14646) Added MetadataServiceMain and RouterServiceMain for the K8s

### PR DESCRIPTION
**Pull request by chtyim**

### CODE REVIEW

File: cdap-master/src/main/java/co/cask/cdap/master/environment/k8s/AppFabricServiceMain.java [hunk 1]
```diff
@@ -121,6 +127,35 @@ protected LoggingContext getLoggingContext() {
                                      Constants.Service.APP_FABRIC_HTTP);
   }
 
+  /**
+   * A Guava {@link Service} for creating storages.
+   */
+  private static final class StorageCreationService extends AbstractIdleService {
```
Comment (anwar6953) [inline cdap-master/src/main/java/co/cask/cdap/master/environment/k8s/AppFabricServiceMain.java:143]: I have a PR which does something similar to `StoreDefinition.createAllTables`, but only with a subset of tables, for the Cloud Runtime case: https://github.com/cdapio/cdap/pull/11081
Would it make sense to use the same service there, but passing in a flag to distinguish whether to create all tables or just the subset that dataset services requires?

Comment (chtyim) [reply cdap-master/src/main/java/co/cask/cdap/master/environment/k8s/AppFabricServiceMain.java:143]: For the one in #11081, I think it is a special workaround, until we removed the reliance on DatasetService entirely. The code change here though, is to stay.

Comment (anwar6953) [reply cdap-master/src/main/java/co/cask/cdap/master/environment/k8s/AppFabricServiceMain.java:143]: Makes sense. So in that PR, I won't try to reuse this class.

**Review by anwar6953 — APPROVED** [review]

**Comment by chtyim** [comment]
Build https://builds.cask.co/browse/CDAP-DUT6688

**Pull request merged**